core/src/mindustry/entities/abilities/StatusFieldAbility.java

```diff
@@ -36,7 +37,7 @@ public String localized(){
     public void update(Unit unit){
         timer += Time.delta;
 
-        if(timer >= reload){
+        if(timer >= reload && (onShoot ? unit.isShooting : !onShoot)){
```

Comment:
Messy and confusing ternary statement, can simply be `!onShoot || unit.isShooting`